docs/building-apps/moneygram-access-integration-guide.mdx

```diff
@@ -247,20 +295,84 @@ In each case, the transaction submitted to Stellar must have a memo attached to
 
 ### Poll Until MoneyGram is Ready
 
-Before the application can send funds or instruct the user to provide cash to a MoneyGram agent, the application should confirm with MoneyGram’s server that the transaction is ready to proceed.
+Before the application can send funds or instruct the user to provide cash to a MoneyGram agent, the application should confirm with MoneyGram’s server that the transaction is ready to proceed which is signaled by the `pending_user_transfer_start` status.
 
 You will need the following information to do so.
 
 - The authentication token provided by MoneyGram
 - The transaction’s ID provided by MoneyGram
-- MoneyGram’s transaction’s endpoint
+- MoneyGram’s transaction’s endpoint (you don't need to worry about it if you are using the Wallet SDK)
   - Testing: https://extstellar.moneygram.com/stellaradapterservice/sep24/transaction
   - Production: https://stellar.moneygram.com/stellaradapterservice/sep24/transaction
 
-This code uses a simple polling mechanism with no bail-out condition. The application’s code should be more robust.
+This code uses a simple watching (polling) mechanism with no bail-out condition. The application’s code should be more robust.
 
 <CodeExample>
 
+```ts
+// We can keep the transaction "id" from the withdraw() call,
+// authToken and assetCode from previous steps.
+const { url, id: transactionId } = await anchor.sep24().withdraw({
+  authToken,
+  assetCode,
+  // ...other params
+});
+
+// First, let's initialize a watcher object from the Wallet SDK.
+let watcher = anchor.sep24().watcher();
+
+// Then we have the option to watch for a particular transaction.
+let { stop, refresh } = watcher.watchOneTransaction({
+  authToken,
+  assetCode,
+  id: transactionId,
+  onMessage: (transaction) => {
+    if (transaction.status === "pending_user_transfer_start") {
+      // begin transfer code
+    }
+  },
+  onSuccess: (transaction) => {
+    // transaction comes back as completed / refunded / expired
+  },
+  onError: (transaction) => {
+    // runtime error, or the transaction comes back as
+    // no_market / too_small / too_large / error
+  },
+});
+
+// We also have the option to watch for ALL transactions of a particular asset.
+let { stop, refresh } = watcher.watchAllTransactions({
+  authToken,
+  assetCode,
+  onMessage: (transaction) => {
+    if (transaction.status === "pending_user_transfer_start") {
+      // begin transfer code
+    }
+  },
+  onError: (transaction) => {
+    // runtime error, or the transaction comes back as
+    // no_market / too_small / too_large / error
+  },
+});
+
+// While the Watcher class offers powerful tracking capabilities, sometimes
+// it's required to just fetch a transaction (or transactions) once. The Anchor
+// class allows you to fetch a transaction by ID, Stellar transaction ID, or
+// external transaction ID:
+const transaction = await anchor.sep24().getTransactionBy({
```

Comment:
Can you expand a bit more into how to use stellarTransactionId or externalTransactionId? Maybe write 3 examples

Reply:
https://github.com/stellar/stellar-docs/pull/658/commits/d45f9d05bb7f78811a3ae68c7849eb6f56ea97a3